### PR TITLE
add initializeAsync for Db implementations

### PR DESCRIPTION
**Pull request by rzpt**
**Goals (and why)**:
There is a cassandra impl of initializeAsync, but not one for DbKvs. If a service depends on this, then it will crash if the db is not up at initialization time.

**Implementation Description (bullets)**:
Copied the cassandra pattern into DbKvs and InDbTimestampBoundStore.
Kept all old function signatures intact to prevent api breaks.

**Testing (What was existing testing like?  What have you done to improve it?)**:
I updated an internal service to use this. I started it with the db off. I waited a few minutes, then turned the db on. The service became healthy.

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

**Review by sudiksha27 — COMMENTED** [review]
Looks correct and has been smoke tested by @rzpt. 
I had some questions around refactoring + it might be worth adding some tests (prod validation using internal testing stack shoul be enough if tests take too long/ are too difficult to write).

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKeyValueService.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+package com.palantir.atlasdb.keyvalue.dbkvs.impl;
+
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.processors.AutoDelegate;
+
+@AutoDelegate
+public interface DbKeyValueService extends KeyValueService {
+
+    String getTablePrefix();
```
[the remark this answers is not in the document]
Comment (rzpt) [reply atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKeyValueService.java:25]: I looked at the usage of the public members internally, and the only one used that I found was checkDatabaseVersion, but that was already done through casting of the object to DbKvs.

**Comment by sudiksha27** [comment]
LGTM! Can merge once we have the changelog + build passes.

**Comment by Jolyon-S** [comment]
you just need to merge develop to fix your build

**Review by sudiksha27 — APPROVED** [review]
👍  LGTM!
@jeremyk-91 please take a look when you can.

**Review by Jolyon-S — COMMENTED** [review]
Just did a quick pass over this to see if anything stands out - as far as I can tell, looks good (although I'm not massively familiar with this part of the code).

**Review by jeremyk-91 — APPROVED** [review]
🕐 0:06 🏁 8:03
👍

**Pull request merged**